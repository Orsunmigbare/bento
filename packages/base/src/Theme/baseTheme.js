import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
