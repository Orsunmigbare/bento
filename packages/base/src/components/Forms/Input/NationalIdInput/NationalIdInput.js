import React from "react"
import Input, { InputPropTypes } from "../Input"

const masks = {
  nb: [
    /[0-9]/,
    /[0-9]/,
    /[0-1]/,
    /[0-9]/,
    /[0-9]/,
    /[0-9]/,
    " ",
    /\d/,
    /\d/,
    /\d/,
    /\d/,
    /\d/
  ]
}

const NationalIdInput = ({ mask, locale = "nb", ...otherProps }) => (
  <Input type={"tel"} mask={masks[locale]} {...otherProps} />
)

NationalIdInput.propTypes = InputPropTypes

export default NationalIdInput
