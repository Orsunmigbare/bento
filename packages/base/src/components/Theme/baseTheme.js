export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
