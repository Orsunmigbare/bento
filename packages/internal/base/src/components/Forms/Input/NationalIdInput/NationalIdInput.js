import React from "react"
import PropTypes from "prop-types"
import Input, { InputPropTypes } from "../Input"

const masks = {
  nb: [
    /[0-9]/,
    /[0-9]/,
    /[0-1]/,
    /[0-9]/,
    /[0-9]/,
    /[0-9]/,
    " ",
    /\d/,
    /\d/,
    /\d/,
    /\d/,
    /\d/
  ]
}
/**
 * Input for National ID Numbers.
 */
const NationalIdInput = ({ mask, id, guide, locale = "nb", ...otherProps }) => (
  <Input
    type={"tel"}
    mask={masks[locale]}
    id={id}
    guide={guide}
    {...otherProps}
  />
)

NationalIdInput.propTypes = {
  ...InputPropTypes,
  locale: PropTypes.oneOf(["nb"]),
  id: PropTypes.string.isRequired,
  guide: PropTypes.bool
}

NationalIdInput.defaultProps = {
  locale: "nb",
  guide: true
}

export default NationalIdInput
