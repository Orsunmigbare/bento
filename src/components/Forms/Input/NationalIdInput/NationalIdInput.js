import React from "react"
import PropTypes from "prop-types"
import Input, { InputPropTypes } from "../Input"

const options = {
  nb: { blocks: [6, 5], numericOnly: true, delimiter: "Â·" }
}
/**
 * Input for National ID Numbers.
 */
const NationalIdInput = React.forwardRef(
  ({ mask, id, guide, locale = "nb", ...otherProps }, ref) => (
    <Input
      type={"tel"}
      ref={ref}
      options={options[locale]}
      id={id}
      {...otherProps}
    />
  )
)

NationalIdInput.propTypes = {
  ...InputPropTypes,
  locale: PropTypes.oneOf(["nb"]),
  id: PropTypes.string.isRequired,
  guide: PropTypes.bool
}

NationalIdInput.defaultProps = {
  locale: "nb",
  guide: false
}

export default NationalIdInput
