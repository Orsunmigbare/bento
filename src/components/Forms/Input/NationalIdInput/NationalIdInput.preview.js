import React from "react"
import NationalIdInput from "./NationalIdInput"
import Toggle from "../../Toggle/Toggle"
import Label from "../../Label/Label"

const preview = {
  title: "NationalIdInput",
  category: "Components/Forms/Input",
  component: NationalIdInput,
  render: props => (
    <NationalIdInput label={"NationalId Input"} id="gfdjrtte35" {...props} />
  ),
  renderExampleController: ({ setComponentState }) => (
    <React.Fragment>
      <Label for="guidePhoneInput">Toggle guides</Label>
      <Toggle
        id="guidePhoneInput"
        onChange={e => setComponentState({ guide: e.target.checked })}
      >
        Guide
      </Toggle>
    </React.Fragment>
  )
}

export default preview
