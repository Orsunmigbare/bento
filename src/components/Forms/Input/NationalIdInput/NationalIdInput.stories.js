import React from "react"
import { storiesOf } from "@storybook/react"
import NationalIdInput from "./NationalIdInput"

storiesOf("components|base/Forms/Input/NationalIdInput", module).add(
  "no props",
  () => {
    return <NationalIdInput label={"NationalIdInput"} id="9852145" />
  }
)
