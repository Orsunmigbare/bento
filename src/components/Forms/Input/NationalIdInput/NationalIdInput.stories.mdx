import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import NationalIdInput from "./NationalIdInput"

<Meta title="Components|Forms/Input/NationalIdInput" component={NationalIdInput}/>


# NationalIdInput
This components controlls user input so that they can only enter numbers with the format of a national id number

```javascript
import {NationalIdInput} from "@staccx/bento"
```

## Standard NationalIdInput
The inputfield only allowed numbers and formats the input with 5-4 and max 9 numbers

<Preview>
  <Story name="Standard">
   <NationalIdInput id="9852145" />
  </Story>
</Preview>




## Props for NationalIdInput
<Props of={NationalIdInput} />
