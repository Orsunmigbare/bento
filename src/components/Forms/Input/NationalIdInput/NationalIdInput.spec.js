import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../../theming/themes/baseTheme"
import NationalIdInput from "./NationalIdInput"
import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"

describe("Alert", () => {
  describe("Snapshots", () => {
    it("ID", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <NationalIdInput id="test" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})

describe("Rendering", () => {
  it("Should render", () => {
    render(
      <ThemeProvider theme={baseTheme}>
        <NationalIdInput id="test" data-testid="testID" />
      </ThemeProvider>
    )
    expect(screen.getByTestId("testID")).toBeInTheDocument()
  })
})
