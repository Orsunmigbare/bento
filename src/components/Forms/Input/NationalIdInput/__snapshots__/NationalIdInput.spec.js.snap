// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Snapshots ID 1`] = `
<div
  className="sc-AxheI jmeqRy"
>
  
  
  <input
    autoFocus={null}
    className="sc-Axmtr bDDSAq"
    disabled={false}
    hidden={false}
    id="test"
    name={null}
    onBlur={null}
    onChange={null}
    onKeyDown={null}
    placeholder={null}
    style={null}
    type="tel"
  />
</div>
`;
